ramips: add support for TP-Link Archer MR200v5

icecream42 opened the pull request:
Specifications:
	
```
CPU:     MT7628AN 580MHZ
RAM:     64MB DDR2
FLASH:   8MB EN25QH64 NOR SPI
WIFI:    2.4GHz 2x2 MT7628 b/g/n internal
WIFI:    5GHz 1x1 MT7610E ac/n PCI
LTE:     Qualcomm MDM9207
ETH:     4xLAN 100base-T integrated
SWITCH:  RT3050-ESW Port 0,1,2,3: LAN, Port 6: CPU
LEDS:    LAN, WAN, Power, 3x signal strength, WiFi
BTNS:    Reset, WiFi toggle
UART:    Near ETH ports, Vcc-GND-RX-TX, 115200, 8N1
```

Installation:

```
1. Update using recovery mode 

 - set your IP to 192.168.0.225, subnet mask: 255.255.255.0 
 - start tftp server, rename tftp-recovery.bin to tp_recovery.bin and place it into the server's directory 
 - while holdig "reset" button, power on the device 
 - keep holding "reset" until the file is being transferred
```

Notes:

```
This board has only one MAC address programmed in the "romfile" partition:

- MAC for phy0 (2.4GHz) at romfile 0xf100 (0)
- MAC for phy1 (5GHz) at romfile 0xf100 (-1)
- stock firmware re-uses phy0 MAC for ethernet
- stock firmware uses romfile 0xf100 (1) for WWAN; not used since QMI interface is raw IP
```

A lot of thanks to the OpenWrt forum for providing important and necessary help, especially gurangax, mrhaav & AndrewZ.

---

Some nitpicks where I wasn't exactly sure what is correct:

1. According to the flash's datasheet the maximum frequency for FAST_READ is 104MHz which I've used in the .dts. Does MT7628AN support this? I unfortunately couldn't figure this out on my own :(
2. I wasn't able to set the MAC for WWAN; is it correct that it's not relevant for a modem in raw IP mode?

musashino205 commented:
> ramips: adding support for TP-Link Archer MR200v5

Please use ```add``` instead of ```adding```.

https://openwrt.org/submitting-patches

> Submission Guidelines
> Submissions should follow the following guidelines:
>
> - write commit subject and message in the imperative form: “add support for X”, NOT “added support for X”

musashino205 commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -606,6 +606,22 @@ define Device/tplink_archer-c50-v6
+  DEVICE_PACKAGES := kmod-mt76x0e uqmi kmod-usb2 kmod-usb-serial-option
+  IMAGES := sysupgrade.bin tftp-recovery.bin
+  IMAGE/tftp-recovery.bin := pad-extra 128k | $$(IMAGE/factory.bin)
+  SUPPORTED_DEVICES := tplink,archer-mr200-v5
```
This will be added automatically by [```Device/Init```](https://github.com/openwrt/openwrt/blob/a0962e8c0435c5511059206c40dbab9dc522767c/include/image.mk#L438) and you can remove this line.

musashino205 commented on target/linux/ramips/dts/mt7628an_tplink_archer-mr200-v5.dts:
```diff
@@ -0,0 +1,190 @@
+				};
+			};
+
+			radio: partition@7f0000 {
```
Please remove unused ```radio: ``` label.

musashino205 commented on target/linux/ramips/dts/mt7628an_tplink_archer-mr200-v5.dts:
```diff
@@ -0,0 +1,190 @@
+				read-only;
+			};
+
+			romfile: partition@7e0000 {
```
Please remove unused ```romfile: ``` label.

musashino205 commented on target/linux/ramips/dts/mt7628an_tplink_archer-mr200-v5.dts:
```diff
@@ -0,0 +1,190 @@
+				reg = <0x20000 0x7b0000>;
+			};
+
+			config: partition@7d0000 {
```
ditto

rmilecki commented on target/linux/ramips/dts/mt7628an_tplink_archer-mr200-v5.dts:
```diff
@@ -0,0 +1,190 @@
+#include "mt7628an.dtsi"
```
SPDX header with licensing info is needed. Ideally it would be:
```
// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
```

musashino205 replied:
I don't see that in the current changes...

icecream42 committed:
ramips: add support for TP-Link Archer MR200v5

Specifications:
	CPU:     MT7628AN 580MHz
	RAM:     64MB DDR2
	FLASH:   8MB EN25QH64 NOR SPI
	WIFI:    2.4GHz 2x2 MT7628 b/g/n internal
	WIFI:    5GHz 1x1 MT7610E ac/n PCI
	LTE:     Qualcomm MDM9207
	ETH:     4xLAN 100base-T integrated
	SWITCH:  RT3050-ESW Port 0,1,2,3: LAN, Port 6: CPU
	LEDS:    LAN, WAN, Power, 3x signal strength, WiFi
	BTNS:    Reset, WiFi toggle
	UART:    Near ETH ports, Vcc-GND-RX-TX, 115200, 8N1

Installation:
	1. Update using recovery mode
        - set your IP to 192.168.0.225, subnet mask: 255.255.255.0
        - start tftp server, rename tftp-recovery.bin to
          tp_recovery.bin and place it into the server's directory
        - while holdig the "reset" button, power on the device
        - keep holding "reset" until the file is being transferred

Notes:
	This board has only one MAC address programmed
        in the "romfile" partition:
        - MAC for phy0 (2.4GHz) at romfile 0xf100 (0)
        - MAC for phy1 (5GHz) at romfile 0xf100 (-1)
        - stock firmware re-uses phy0 MAC for ethernet
        - stock firmware uses romfile 0xf100 (1) for WWAN;
          not used since QMI interface is raw IP

Signed-off-by: Lea Teuberth <lea.teuberth@outlook.com>

hauke commented on target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds:
```diff
@@ -83,6 +83,10 @@ tplink,archer-c50-v6)
 	ucidef_set_led_wlan "wlan5g" "wlan5g" "green:wlan5g" "phy1tpt"
 	;;
+tplink,archer-mr200-v5)
+	ucidef_set_led_netdev "lan" "lan" "white:lan" "eth0.1"
```
Why do you trigger the LED on `eth0.1` and not `eth0`?
The `.1` will link it to a vlan interface.
